Include/internal/pycore_bitutils.h

```diff
@@ -131,8 +131,40 @@ _Py_popcount32(uint32_t x)
 }
 
 
+// Return the index of the most significant 1 bit in 'x'. This is the smallest
+// integer k such that x < 2**k. Equivalent to floor(log2(x)) + 1 for x != 0.
+static inline int
+_Py_bit_length(unsigned long x) {
+    if (x == 0) {
+        return 0;
+    }
+#if (defined(__clang__) || (defined(__GNUC__) && (((__GNUC__ == 3) && (__GNUC_MINOR__ >= 4)) || (__GNUC__ >= 4))))
+    // Undefined behavior for x == 0.
+    return sizeof(unsigned long) * 8 - __builtin_clzl(x);
+#elif defined(_MSC_VER)
+    Py_BUILD_ASSERT(sizeof(unsigned long) <= 4);
+    // Does not write to msb if x == 0.
+    unsigned long msb;
+    _BitScanReverse(&msb, x);
```

Comment:
Maybe use the return value? Something like:

```
if (_BitScanReverse(&msb, x)) {
    return (int)msb + 1;
}
else {
    return 0;
}
```

In this case, move the `if (x == 0) ...` in ther other #if cases.